builtin/providers/aws/resource_aws_instance.go

```diff
@@ -749,13 +892,36 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 	return resourceAwsInstanceRead(d, meta)
 }
 
+// FIXME add test to ensure spot instances are deleted
 func resourceAwsInstanceDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
+	if reqID := d.Get("spot_request_id").(string); reqID != "" {
+		log.Printf("[INFO] Cancelling spot request: %s", reqID)
+		output, err := conn.CancelSpotInstanceRequests(&ec2.CancelSpotInstanceRequestsInput{
+			SpotInstanceRequestIDs: []*string{aws.String(reqID)},
+		})
+
+		if err != nil {
+			return fmt.Errorf(
+				"Error cancelling spot reservation (%s): %s",
+				reqID, err)
+		}
+
+		if output != nil {
+			cancelled := output.CancelledSpotInstanceRequests
+			if len(cancelled) != 1 || *cancelled[0].SpotInstanceRequestID != reqID {
```

Comment:
I'm a little wary of this logic myself. It might cause the stack to fail to delete if the spot instance request wasn't fulfilled correctly (I'm not sure how cancel responds to requests that are already cancelled / in some other bad state)

What do you guys think we should do here?